llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -218,8 +218,11 @@ void collectCompositeElementsInfoRecursive(
     const Module &M, Type *Ty, const ID *&IDIter, unsigned &Offset,
     std::vector<SpecConstantDescriptor> &Result) {
   if (IDIter->Undef) {
-    // We can just skip undefined values because every such value is just a
-    // padding and will be handled in a different manner.
+    // Such ID is expected to be a padding within a non-packed struct, we need
+    // to move offset to account for it.
+    // Offset += M.getDataLayout().getTypeSizeInBits(Ty) * /* bits in byte */ 8;
```

Comment:
Should probably be removed (at least partially). I know that I'm probably the author of those, but I need a moment to get back up to speed with all the details here to suggest anything concrete